sound/soc/sof/sof-acpi-dev.c

```diff
@@ -24,6 +24,14 @@
 /* platform specific devices */
 #include "intel/shim.h"
 
+static char *fw_path;
+module_param(fw_path, charp, 0444);
+MODULE_PARM_DESC(fw_path, "alternate path for SOF firmware.");
+
+static char *tplg_path;
+module_param(tplg_path, charp, 0444);
+MODULE_PARM_DESC(tplg_path, "alternate path for SOF topology.");
```

Comment:
Wouldn't it be better to have these parameters in a common sof module instead of repeating them in each interface module? We don't expect to load several of them simultaneously and even if we did, letting them have different firmware and topology paths seems like being too kind to the user to me ;-)

Reply:
I am not happy with the copy/paste here, but I couldn't think of anything better. soc-acpi-dev and soc-pci-dev are the top-level modules that will setup everything, I don't know how we can introduce a lower-level common module.

Reply:
I ended-up moving the filename handling to lower levels, and deal with the module parameters only in sof-pci/acpi/spi-dev. that factors in more code and gives us more opportunities for quirks, e.g. for Hdaudio which works fine now.